### ISSUE TITLE
move not remove files from the source that already exist in the destination

### ISSUE THREAD

**Issue by RXWatcher1** [ncw/rclone#751]
When filing an issue, please include the following information if
possible as well as a description of the problem.  Make sure you are
using the [latest version of rclone](http://rclone.org/downloads/).

> What is your rclone version (eg output from `rclone -V`)
rclone v1.33-13-g7be4c3a-fuse-flagsβ

> Which OS you are using and how many bits (eg Windows 7, 64 bit)
Ubuntu 16.04
> Which cloud storage system are you using? (eg Google Drive)
Encrypted Amazon Cloud Drive
> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
rclone move /tv_shows/ eacd:media/tv_shows -v

> A log from the command with the `-v` flag (eg output from `rclone -v copy /tmp remote:tmp`)

2016/09/29 16:45:11 Encrypted amazon drive root '40vr5l7bbib53ql2gesinnck6c/lnhbggu7s6ai3oora37epq5qmk': Waiting for checks to finish
2016/09/29 16:45:11 Ink Master/folder.jpg: Sizes identical
2016/09/29 16:45:11 Ink Master/folder.jpg: Unchanged skipping
2016/09/29 16:45:11 Ink Master/logo.png: Sizes identical
2016/09/29 16:45:11 Ink Master/landscape.jpg: Sizes identical
2016/09/29 16:45:11 Ink Master/landscape.jpg: Unchanged skipping
2016/09/29 16:45:11 Ink Master/fanart.jpg: Sizes identical
2016/09/29 16:45:11 Ink Master/tvshow.nfo: Sizes identical
2016/09/29 16:45:11 Ink Master/tvshow.nfo: Unchanged skipping
2016/09/29 16:45:11 Ink Master/fanart.jpg: Unchanged skipping
2016/09/29 16:45:11 Ink Master/logo.png: Unchanged skipping
2016/09/29 16:45:11 Ink Master/S8/folder.jpg: Sizes identical
2016/09/29 16:45:11 Ink Master/S8/folder.jpg: Unchanged skipping
2016/09/29 16:45:11 Ink Master/clearart.png: Sizes identical
2016/09/29 16:45:11 Ink Master/clearart.png: Unchanged skipping
2016/09/29 16:45:11 Ink Master/S8/poster.jpg: Sizes identical
2016/09/29 16:45:11 Ink Master/S8/poster.jpg: Unchanged skipping
2016/09/29 16:45:11 Ink Master/poster.jpg: Sizes identical
2016/09/29 16:45:11 Ink Master/poster.jpg: Unchanged skipping
2016/09/29 16:45:11 Encrypted amazon drive root '40vr5l7bbib53ql2gesinnck6c/lnhbggu7s6ai3oora37epq5qmk': Waiting for transfers to finish
2016/09/29 16:45:11
Transferred:      0 Bytes (0 Bytes/s)
Errors:                 0
Checks:                 9
Transferred:            0
Elapsed time:        7.8s
2016/09/29 16:45:11 Go routines at exit 33

It had issues with uploading some of these initially but they seem to have uploaded but it will not remove them from the local directory on a second run. I noticed this on other files as well, if I did a 'move' it would find the ones in ACD and then say the size was identical and do the unchanged skipping but then leave them in the local directory.

I expected that it would remove the local copies since I did the 'move' option. Is that not the case?

└── tv_shows
    └── Ink Master
        ├── clearart.png
        ├── fanart.jpg
        ├── folder.jpg
        ├── landscape.jpg
        ├── logo.png
        ├── poster.jpg
        ├── S8
        │   ├── folder.jpg
        │   └── poster.jpg
        └── tvshow.nfo

Thank you!

Edit: adding -I to the command uploaded the files again and deleted the local copies.

**Comment by xor-zz** [comment]
I noticed the same behavior and believe it is intentional but undocumented that a move that doesn't actually involve a copy isn't destructive.

http://rclone.org/commands/rclone_move/

Notes that

> If possible a server side move will be used, otherwise it will copy it (server side if possible) into dest:path then delete the original (if no errors on copy) in source:path.

I believe this should be

> If possible a server side move will be used, otherwise it will copy it (server side if possible) into dest:path then delete the original (if the file was moved) in source:path.

To cover the case where you skip copying of the local copy because it matches the already uploaded file.

**Comment by RXWatcher1** [comment]
It seems like odd behavior because it's a 'move'. If it encounters the same file on the destination(size, etc match), then the intent of the 'move' should remove that file from the source.  I don't want to have to use the -I all of the time to get the behavior I need because it's forcing the re-transferring of data that is already good. It's wasting resources. Maybe we need another option that is destructive if this was done intentionally.

**Comment by xor-zz** [comment]
I don't actually see the -l argument documented either. What is the command supposed to do?

I think the closest analog is rsync with "--remove-source-files" which does function as would be expected - if the source files already exist at the destination they are removed, even if they don't need to be copied.

**Comment by RXWatcher1** [comment]
I found it in the rclone --help.

-I, --ignore-times                    Don't skip files that match size and time - transfer all files
 
so its transferring  the files again. This isnt a big deal on small jpegs but it is a big deal on Multi-GB files. This will transfer the file again and clean it up on the source.

I know the intended behavior is to leave directories behind so I clean those up with this after the move. 
find /path/to/source -empty -type d -delete

I noticed it wasnt cleaning up the directories and thats when I saw that there were files left behind.

**Comment by xor-zz** [comment]
#754 updates the documentation to reflect this behavior.

**Comment by ncw** [comment]
I think this is a bug - `move` should move the files and if an identical file exists in the destination then the source file should be deleted.  This is what would happen if it was copied

Here is a demo of the problem.

```
$ mkdir /tmp/a /tmp/b
$ echo hellp >/tmp/a/test.txt
$ rclone -q copy /tmp/a /tmp/b
$ rclone move -v /tmp/a /tmp/b
2016/10/01 17:08:33 rclone: Version "v1.33-DEV" starting with parameters ["rclone" "move" "-v" "/tmp/a" "/tmp/b"]
2016/10/01 17:08:33 Local file system at /tmp/b: Modify window is 1ns
2016/10/01 17:08:33 Local file system at /tmp/b: Using server side directory move
2016/10/01 17:08:33 Local file system at /tmp/b: Server side directory move failed - fallback to file moves: can't copy directory - destination already exists
2016/10/01 17:08:33 Local file system at /tmp/b: Waiting for checks to finish
2016/10/01 17:08:33 test.txt: Size and modification time the same (differ by 0s, within tolerance 1ns)
2016/10/01 17:08:33 test.txt: Unchanged skipping
2016/10/01 17:08:33 Local file system at /tmp/b: Waiting for transfers to finish
2016/10/01 17:08:33 
Transferred:      0 Bytes (0 Bytes/s)
Errors:                 0
Checks:                 1
Transferred:            0
Elapsed time:          0s
2016/10/01 17:08:33 Go routines at exit 3
$ rclone -q ls /tmp/a
        6 test.txt
```

**Comment by xor-zz** [comment]
I think the behavior should be changed as well.

At a minimum, if it is not changed, the documentation needs to be updated.

**Comment by ncw** [comment]
I've fixed this in this beta: http://pub.rclone.org/v1.33-30-g945f49a%CE%B2/

If you find any problems then please re-open the ticket!

**Comment by ncw** [comment]
Arg - don't use that beta - I brown paper bagged it!  I'll post another in a moment

**Comment by ncw** [comment]
http://pub.rclone.org/v1.33-31-g9ea20ba%CE%B2/

**Comment by im7mortal** [comment]
I did an extensive search to figure out how to remove files from the source after I had already run sync.

Here is my scenario:

1. I have 100 AWS buckets with around 30 million small objects each.
2. Using AWS Transfer services was too much of a hassle.
3. I prepared a Python script that generated rclone sync commands. I ran them and validated the results. There were some non-important transfer failures due to AWS ACL issues.
4. Now I want to clean up the source. I would still like a chance to validate everything one more time, so I was thinking of using rclone move, where identical files in the source and destination would be removed from the source.
5. However, I discovered that rclone move does not remove files if they were not actually moved.

I tried different flags, but I never succeeded.
